modules/core/src/lib/deck.ts

```diff
@@ -248,7 +246,6 @@ const defaultProps = {
   device: null,
   deviceProps: {type: 'webgl'} as DeviceProps,
```

Comment:
This will not provide any type hint for WebGL context options. Should be something like this?

```ts
DeviceProps & (
  ({type: 'webgl'} & WebGLContextAttributes)
  | {type: 'webgpu'}
)
```

Reply:
You may want to specify `preserveDrawingBuffer` even when omitting Device type or using `type: 'best-available'`.

Reply:
Is `type` optional? What happens when someone doesn't specify the `type` on `deviceProps`, and let's deck define the device? 

What will luma.createDevice do by default in deck? I think deck only registers a WebGLDevice in this logic but it's just a little unclear reading the code whether type is required, and if not, what that implies in deck?